client/src/modules/StudentDashboard/hooks/useSubmitTaskSolution.ts

```diff
@@ -0,0 +1,71 @@
+import { AxiosError } from 'axios';
+import { useReducer } from 'react';
+import { CourseTaskDto, CoursesTasksApi, CreateCourseTaskDtoCheckerEnum, CoursesTaskSolutionsApi } from 'api';
+import { stageInterviewType } from '../../../domain/interview';
+
+type Action = {
+  type: 'loading' | 'open' | 'close' | 'error' | 'submit';
+  state?: State;
+};
+
+export type State = {
+  errorText?: string;
+  submitted?: boolean;
+  data?: { courseTasks: CourseTaskDto[] };
+  loading?: boolean;
+} | null;
+
+function reducer(state: State, action: Action): State {
+  switch (action.type) {
+    case 'loading':
+      return { loading: true };
+    case 'open':
+      return { loading: false, data: action.state?.data };
+    case 'submit':
+      return { submitted: true, data: state?.data };
+    case 'close':
+      return null;
+    case 'error':
+      return { errorText: action.state?.errorText, data: state?.data };
+    default:
+      return state;
+  }
+}
+
+export function useSubmitTaskSolution(courseId: number) {
+  const [state, dispatch] = useReducer(reducer, null);
+
+  const showModal = async () => {
+    dispatch({ type: 'loading' });
+    const coursesTasksApi = new CoursesTasksApi();
+    const { data } = await coursesTasksApi.getCourseTasks(courseId);
+    const courseTasks = data.filter(
+      item =>
+        item.checker === CreateCourseTaskDtoCheckerEnum.Mentor &&
+        item.type != 'selfeducation' &&
```

Comment:
please check if we have type enum in `api`. If so, I suggest to use it

Reply:
Fixed, thank you